src/ImageSharp/Metadata/Profiles/Exif/ExifReader.cs

```diff
@@ -275,139 +288,134 @@ private object ConvertValue(ExifDataType dataType, ReadOnlySpan<byte> buffer, ui
-
-                    this.position = oldIndex;
-                    return false;
-                }
-
-                this.TryReadSpan((int)size, out ReadOnlySpan<byte> dataBuffer);
-
-                value = this.ConvertValue(dataType, dataBuffer, numberOfComponents);
-                this.position = oldIndex;
+                    var dataBuffer = new byte[size];
+                    this.Seek(newOffset);
+                    if (this.TryReadSpan(dataBuffer))
+                    {
+                        object value = this.ConvertValue(dataType, dataBuffer, numberOfComponents);
+                        this.Add(values, exifValue, value);
+                    }
+                });
             }
             else
             {
-                value = this.ConvertValue(dataType, offsetBuffer, numberOfComponents);
-            }
+                object value = this.ConvertValue(dataType, this.offsetBuffer, numberOfComponents);
 
-            exifValue = ExifValues.Create(tag) ?? ExifValues.Create(tag, dataType, numberOfComponents);
+                this.Add(values, exifValue, value);
+            }
+        }
 
-            if (exifValue is null)
+        private void Add(IList<IExifValue> values, IExifValue exif, object value)
+        {
+            if (!exif.TrySetValue(value))
             {
-                this.AddInvalidTag(new UnkownExifTag(tag));
-                return false;
+                return;
             }
 
-            if (!exifValue.TrySetValue(value))
+            foreach (IExifValue val in values)
             {
-                return false;
+                // sometimes duplicates appear,
+                // can compare val.Tag == exif.Tag
+                if (val == exif)
+                {
+                    Debug.WriteLine($"Duplicate Exif tag: tag={exif.Tag}, dataType={exif.DataType}");
+                    return;
+                }
             }
 
-            return true;
+            if (exif.Tag == ExifTag.SubIFDOffset)
+            {
+                this.exifOffset = (uint)value;
+            }
+            else if (exif.Tag == ExifTag.GPSIFDOffset)
+            {
+                this.gpsOffset = (uint)value;
+            }
+            else
+            {
+                values.Add(exif);
+            }
         }
 
         private void AddInvalidTag(ExifTag tag)
-            => (this.invalidTags ?? (this.invalidTags = new List<ExifTag>())).Add(tag);
+            => (this.invalidTags ??= new List<ExifTag>()).Add(tag);
+
+        private void Seek(long pos)
+            => this.data.Seek(pos, SeekOrigin.Begin);
 
-        private bool TryReadSpan(int length, out ReadOnlySpan<byte> span)
+        private bool TryReadSpan(Span<byte> span)
         {
-            if (this.RemainingLength < length)
+            int length = span.Length;
+            if ((this.data.Length - this.data.Position) < length)
             {
                 span = default;
```

Comment:
Seems odd here to be assigning a value to the span.

Reply:
Removed